### PR TITLE
feat: query logs api

### PR DESCRIPTION
**Pull request by DefK1lla**

### CODE REVIEW

File: src/components/QueryFilter/index.tsx [hunk 1]
```diff
@@ -4,23 +4,38 @@ import { useForm } from "react-hook-form";
...
-  dataSources: string[];
-  onChange: (data: DeepPartial<QueryFilterForm>) => void;
+  dataSources: DataSourceInfo[];
+  defaultValues?: Partial<QueryFilterForm>;
+  values?: QueryFilterForm;
+  onChange: (data: QueryFilterForm) => void;
 }
 
-const QueryFilter: FC<QueryFilterProps> = ({ dataSources, onChange }) => {
+const QueryFilter: FC<QueryFilterProps> = ({
```
Comment (ifokeev) [inline src/components/QueryFilter/index.tsx:20]: `QueryFilters`

File: src/components/QueryLogsTable/index.stories.tsx [hunk 2]
```diff
@@ -19,5 +19,1691 @@ const Template: StoryFn<typeof QueryLogsTable> = (args) => (
...
+          aggregate: undefined,
+          nodes: [],
+        },
+        updated_at: undefined,
+        versions: [],
+        versions_aggregate: {
+          __typename: undefined,
+          aggregate: undefined,
+          nodes: [],
+        },
```
Comment (ifokeev) [inline src/components/QueryLogsTable/index.stories.tsx:84]: all empty fields could be removed

File: src/components/QueryLogsTable/index.stories.tsx [hunk 3]
```diff
@@ -19,5 +19,1691 @@ const Template: StoryFn<typeof QueryLogsTable> = (args) => (
...
+      request_event_logs: [
+        { path: "/cubejs/datasources/v1/meta", error: null },
+        { path: "/cubejs/datasources/v1/meta", error: null },
+      ],
+      request_event_logs_aggregate: {
+        aggregate: { count: 2 },
+        nodes: [],
+      },
+    },
+  ],
```
Comment (ifokeev) [inline src/components/QueryLogsTable/index.stories.tsx:1708]: please keep only valuable data

File: config/routes.ts [hunk 4]
```diff
@@ -47,6 +47,10 @@ export default [
             path: "/settings/info",
             component: "./pages/PersonalInfo",
           },
+          {
+            path: "/logs/query",
```
Comment (ifokeev) [inline config/routes.ts:51]: should be link in the menu too

**Review by Libertonius — CHANGES_REQUESTED** [review]

File: src/pages/QueryLogs/index.tsx [hunk 5]
```diff
@@ -34,11 +85,76 @@ const QueryLogs: React.FC<QueryLogsProps> = ({ logs }) => {
...
+    </SidebarLayout>
+  );
+};
+
+const QueryLogsWrapper = () => {
+  const [filter, setFilter] = useState(defaultFilterState);
+
+  const { withAuthPrefix } = useAppSettings();
+  const [, setLocation] = useLocation();
+  const basePath = withAuthPrefix("/logs/requests");
```
Comment (ifokeev) [inline src/pages/QueryLogs/index.tsx:119]: why url is `/logs/requests/` but page url is `/logs/query`? Base path should be the same

File: src/components/QueryFilters/index.tsx [hunk 6]
```diff
@@ -3,24 +3,39 @@ import { useTranslation } from "react-i18next";
...
-  onChange: (data: DeepPartial<QueryFilterForm>) => void;
+interface QueryFiltersProps {
+  dataSources?: DataSourceInfo[];
+  defaultValues?: Partial<QueryFiltersForm>;
+  values?: QueryFiltersForm;
+  onChange: (data: QueryFiltersForm) => void;
 }
 
-const QueryFilter: FC<QueryFilterProps> = ({ dataSources, onChange }) => {
+const QueryFilters: FC<QueryFiltersProps> = ({
```
Comment (ifokeev) [inline src/components/QueryFilters/index.tsx:20]: date input is not keeping selected value, should keep on select

File: src/components/QueryLogsTable/index.tsx [hunk 7]
```diff
@@ -91,7 +100,11 @@ const QueryLogsTable: FC<QueryLogsTableProps> = ({ logs }) => {
       columns={columns}
       dataSource={logs}
       rowKey={(record) => record.id}
-      pagination={false}
+      onRow={(r) => ({
```
Comment (ifokeev) [inline src/components/QueryLogsTable/index.tsx:103]: on row hover should have `cursor: pointer`, because it's clickable

**Review by ifokeev — CHANGES_REQUESTED** [review]

**Review by ifokeev — APPROVED** [review]

**Review by Libertonius — APPROVED** [review]

**Pull request merged**